src/acvp_rsa.c

```diff
@@ -332,6 +438,74 @@ acvp_prep_keygen(int rand_pq, unsigned int mod, int info_gen_by_server,
     return rv;
 }
 
+static ACVP_RESULT acvp_kat_rsa_sig(unsigned int tc_id, ACVP_CIPHER alg_id,
+        JSON_Object* groupobj, JSON_Object* testobj, ACVP_CAPS_LIST* cap,
+        ACVP_CTX* ctx, ACVP_RSA_SIG_TC* sigtc)
+{
+    ACVP_RESULT rv;
+
+    /*
+     * Sig attrs for group obj
+     */
+    char *sig_type;
+
+
+    /*
+     * siggen attrs for test obj
+     */
+    unsigned int modulo, salt_len;
+    unsigned char *msg;
+    char *hash_alg;
+
+    salt_len = 0;
+
+    /*
+     * sigver attrs for test obj
+     */
+    //they go here!!!
+
+    /*
+     * set group obj -- common to siggen and sigver
+     */
+    sig_type = (char *)json_object_get_string(groupobj, "sigType");
+
+    /*
+     * set test obj
+     */
+    switch(sigtc->mode) {
+    case ACVP_RSA_MODE_SIGGEN:
+    {
+        modulo = (unsigned int)json_object_get_number(groupobj, ACVP_RSA_SIG_MODULO_OBJ_NAME);
+        hash_alg = (char *)json_object_get_string(groupobj, ACVP_RSA_HASHALG_OBJ_NAME);
+        msg = (unsigned char *)json_object_get_string(testobj, ACVP_RSA_SIG_MSG_OBJ_NAME);
+        if(strncmp(sig_type, RSA_SIG_TYPE_PKCS1PSS_NAME, RSA_SIG_TYPE_MAX_LEN ) == 0 ) {
+            salt_len = (unsigned int)json_object_get_number(testobj, ACVP_RSA_SALTLEN_OBJ_NAME);
+        }
+
+        /*
+         * Setup the test case data that will be passed down to
+         * the crypto module.
+         * TODO: this does mallocs,kat we can probably do the mallocs once for
+         *       the entire vector set to be more efficient
+         */
+        rv = acvp_rsa_init_sig_tc(ctx, sigtc, tc_id, alg_id, /* note: mode is set in kat_handler */
+                sig_type, modulo, hash_alg, msg, salt_len); /* siggen attrs */
+        break;
+    }
+//    case ACVP_RSA_MODE_SIGVER:
```

Comment:
Should this case be removed or completed?

Reply:
removed until the next pull request when I work on SigVer test case handling?